src/main/java/com/github/bordertech/config/DefaultConfiguration.java

```diff
@@ -158,27 +149,32 @@
 	// -----------------------------------------------------------------------------------------------------------------
 	// State used during loading of parameters
 	/**
-	 * The messages logged during loading of the configuration. \
+	 * The messages logged during loading of the configuration.
 	 */
 	private final StringBuilder messages = new StringBuilder();
 
-	/**
-	 * The resource being loaded. This is used for the relative form of resource loading.
-	 */
-	private final Deque<String> resources = new ArrayDeque<>();
-
 	/**
 	 * A generic object that allows us to synchronized refreshes. Required so that gets and refreshes are threadsafe
 	 */
 	private final Object lockObject = new Object();
 
 	// -----------------------------------------------------------------------------------------------------------------
 	// Implementation
+	/**
+	 * Resource load order.
+	 */
+	private final String[] resourceLoadOrder;
+
 	/**
 	 * Holds the current environment suffix (if set).
+	 *
+	 * @deprecated - Replaced by current Profile
 	 */
+	@Deprecated
 	private String currentEnvironment = null;
 
+	private String environmentProfile = null;
```

Comment:
Did we want to avoid using `environment` in the name with profile? what if we call it `currentProfile`?

Maybe I am confusing what environment and profile are being used for, but can we remove the `currentEnvironment` variable. I assume that when we are checking a profile has been set we check the new profile key, but if it hasnt been set we also check the legacy environment key. But as we will only use the one value, I assume we dont need to hold both values.